runtime/src/bank.rs

```diff
@@ -7259,6 +7259,13 @@ impl Bank {
             self.get_minimum_balance_for_rent_exemption(account.data().len())
                 .max(account.lamports()),
         );
+
```

Comment:
This may be sufficient for any existing sysvar (although I have yet to audit), but I think there is no guarantee that a new sysvar must use this method. For instance, the epoch rewards partition data account that I added and then reverted would not have been addressed by this change ([link](https://github.com/solana-labs/solana/pull/35088/files#diff-ed47b4a0198313377e091bb3957bbbc63d937805426d1b2b6de39d0a50d32a0cL3624)), as it called `AccountSharedData::new_data()` which ultimately calls `Account::create()` with `rent_epoch: Epoch::default()`.

It would be nice if we could think of a way to make this less brittle for new use-cases. Perhaps we need a method under sysvar that populates an account with the right `rent_epoch` value. I wonder if we need to deprecate existing Account creation methods also (could get complex if that field is reused for something).

I'll think on this more.

Reply:
> It would be nice if we could think of a way to make this less brittle for new use-cases. Perhaps we need a method under sysvar that populates an account with the right rent_epoch value. I wonder if we need to deprecate existing Account creation methods also (could get complex if that field is reused for something).

yes. we expose far too many implementation details in far too many places. i fully support deprecation of the existing interfaces in favor of more a more restrictive one that funnels into a unified code path with as few public details as possible